From 31b7aa3a6b08fb33421e12793348e30fb8466041 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Fri, 14 Sep 2018 21:36:01 +0800
Subject: [PATCH 4717/5242] MLK-19664-2 ARM64: dts: imx8qm domu: enable gpio
 controller

commit  f67189a80d3f48c864380e006c87549010f6c921 from
https://source.codeaurora.org/external/imx/linux-imx.git

Enable the gpio controller, but remove the power-domain property.
Currently we are using mmio trap in XEN to handle GPIO.
For Dom0, it has the highest priviledge to read/write any gpio registers
and the interrupts are also forwarded to Dom0.

For DomU, a new property as following is added to doma node in dom0 dts.
	gpios = <&gpio1 13 GPIO_ACTIVE_LOW>, <&gpio4 9 GPIO_ACTIVE_LOW>;
xen will parse it and restrict domu read/write behavior.

xen,shared means mmio trap is used.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-dom0.dts     |    8 ++++
 .../boot/dts/freescale/fsl-imx8qm-mek-domu.dts     |   42 ++++++++++++++++----
 2 files changed, 42 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
index db9f3f9..06c9acb 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-dom0.dts
@@ -407,6 +407,14 @@
        xen,passthrough;
 };
 
+&gpio1 {
+	xen,shared;
+};
+
+&gpio4 {
+	xen,shared;
+};
+
 &dsp {
        xen,passthrough;
 };
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
index 66cd7f5..a3b47a9 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-domu.dts
@@ -433,14 +433,40 @@
 /delete-node/ &edma0;
 /delete-node/ &edma2;
 /delete-node/ &edma3;
-/delete-node/ &gpio0;
-/delete-node/ &gpio1;
-/delete-node/ &gpio2;
-/delete-node/ &gpio3;
-/delete-node/ &gpio4;
-/delete-node/ &gpio5;
-/delete-node/ &gpio6;
-/delete-node/ &gpio7;
+
+&gpio0 {
+	/delete-property/ power-domains;
+	status = "disabled";
+};
+&gpio1 {
+	/delete-property/ power-domains;
+	status = "okay";
+};
+&gpio2 {
+	/delete-property/ power-domains;
+	status = "disabled";
+};
+&gpio3 {
+	/delete-property/ power-domains;
+	status = "disabled";
+};
+&gpio4 {
+	/delete-property/ power-domains;
+	status = "okay";
+};
+&gpio5 {
+	/delete-property/ power-domains;
+	status = "disabled";
+};
+&gpio6 {
+	/delete-property/ power-domains;
+	status = "disabled";
+};
+&gpio7 {
+	/delete-property/ power-domains;
+	status = "disabled";
+};
+
 /delete-node/ &gpio0_mipi_csi0;
 /delete-node/ &gpio0_mipi_csi1;
 /delete-node/ &gpt0;
-- 
1.7.9.5

